### ISSUE TITLE
bug: float32 data not returned in <pkg>.stress_period_data.to_array()

### ISSUE THREAD

**Issue by dbrakenhoff** [modflowpy/flopy#1779]
**Describe the bug**
I'm not sure whether this is considered a bug, but I came across the following in one of my scripts. I downloaded some online data through an API and used that to build my DRN package. The array dtype was apparently float32, and this is treated as an object dtype in `MFTransientList`. This means that when you convert the `stress_period_data` back to an array with `to_array()` the array containing the float32 data is not returned.

**To Reproduce**
```python
import flopy
import numpy as np

sim = flopy.mf6.MFSimulation()
tdis = flopy.mf6.ModflowTdis(sim)
gwf = flopy.mf6.ModflowGwf(sim)
dis = flopy.mf6.ModflowGwfdis(gwf)

drn_data = [[(0, 0, 0), np.float32(1.0), np.float64(100.0)]]  # note float32 vs float64
drn = flopy.mf6.ModflowGwfdrn(gwf, stress_period_data={0: drn_data})

drn.stress_period_data.to_array()
```
This returns:
```raw
{'cond': array([[[100.,   0.],
                 [  0.,   0.]]])}
```

**Expected behavior**
I'd expect the "elev" data to be returned as well, despite it being float32.

The fix is easy, just convert your input data to float64, but perhaps flopy could do that for me?

**Desktop:**
 - OS: Ubuntu 22.04

Thanks for taking a look!

**Comment by spaulins-usgs** [comment]
Currently FloPy stores DRN elevation in an object dtype since elevation can potentially be specified as a string from a time series file.  While in your example there are no time series files specified, FloPy always uses the same dtype for elevation for consistency.

I agree that supporting multiple types in a single recarray column can be problematic.  We are currently discussing possible future alternatives to this.

**Comment by spaulins-usgs** [comment]
Sorry, I misunderstood the problem.  Yes you are correct that it should return the same result whether you specify elevation as a float32 or float64.  Confirmed that this is a bug.

**Comment by spaulins-usgs** [comment]
@dbrakenhoff, I just added a fix to flopy to better handle float32 numbers.  Please download the FloPy develop branch from github and confirm that this fixes the problem for you.

**Comment by dbrakenhoff** [comment]
@spaulins-usgs, thanks, it's working!

**Comment by w-bonelli** [comment]
closed by #1806
